src/client/components/Meals.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from "react";
+import Meal from "./Meal";
+import images from "./images";
+import common from "../assets/images/common.png";
+
+import { Route, Switch, Link, useRouteMatch } from "react-router-dom";
+
+const Meals = () => {
```

Comment:
Some issue as before. This component does a lot of different things. Remember the difference between imperative and declarative programming. Here you give the computer a recipe for what is to be done, but you are forcing the reader to think and spend time on deciphering the logic. For instance the long ternary statement. Imagine you had to maintain thousands of lines of code like this.